plugins/ec2/crossVpcPublicPrivate.js

```diff
@@ -0,0 +1,222 @@
+var async = require('async');
+var helpers = require('../../helpers');
+
+module.exports = {
+    title: 'Cross VPC Public Private Communication',
+    category: 'EC2',
+    description: 'Ensures communication between public and private VPC tiers is not enabled',
+    more_info: 'Communication between the public tier of one VPC and the private tier of other VPCs should never be allowed. Instead, VPC peerings with proper NACLs and gateways should be used',
+    link: 'https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Subnets.html',
+    recommended_action: 'Remove the NACL rules allowing communication between the public and private tiers of different VPCs',
+    apis: ['EC2:describeSubnets', 'EC2:describeRouteTables', 'EC2:describeVpcPeeringConnections'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+
+
+        async.each(helpers.regions.ec2, function(region, rcb) {
+
+
+                // for Subnets
+                var describeSubnets = helpers.addSource(cache, source, ['ec2', 'describeSubnets', region]);
+
+                // error handling
+                if (!describeSubnets || !describeSubnets.data || describeSubnets.err) {
+                    helpers.addResult(results, 3, 'Unable to query for Subnets: ' + helpers.addError(describeSubnets), region);
+                }
+
+                // grapping a list of Vpcs and Subnets
+                var subVpc = [];
+                for (i in describeSubnets.data) {
+                    subVpc.push({
+                        vcpId: describeSubnets.data[i].VpcId,
+                        subId: describeSubnets.data[i].SubnetId,
+                        subCidr: describeSubnets.data[i].CidrBlock
+                    });
+                }
+
+
+
+                // for RouteTables
+                var describeRouteTables = helpers.addSource(cache, source, ['ec2', 'describeRouteTables', region]);
+
+                // error handling
+                if (!describeRouteTables || !describeRouteTables.data || describeRouteTables.err) {
+                    helpers.addResult(results, 3, 'Unable to query for RouteTables: ' + helpers.addError(describeRouteTables), region);
+                }
+
+
+                // for VpcPeeringConnections
+                var describeVpcPeeringConnections = helpers.addSource(cache, source, ['ec2', 'describeVpcPeeringConnections', region]);
+
+                // error handling
+                if (!describeVpcPeeringConnections || !describeVpcPeeringConnections.data || describeVpcPeeringConnections.err) {
+                    helpers.addResult(results, 3, 'Unable to query for VpcPeering: ' + helpers.addError(describeRouteTables), region);
+                }
+
+                if (!describeVpcPeeringConnections.data.length) {
+                    helpers.addResult(results, 0, 'No public private subnets connection found', region);
```

Comment:
If there are no peering connections, this should `return rcb()` to prevent further execution